src/infil.c

```diff
@@ -308,6 +334,7 @@ double infil_getInfil(int j, int m, double tstep, double rainfall,
 //  Purpose: computes infiltration rate depending on infiltration method.
 //
 {
+    m = Subcatch[j].infil_type; // 2016-01-15: CHANGE PASS-BY-VALUE, m
```

Comment:
@aerispaha, `m` is an input argument so I think the caller of `infil_getInfil` should continue to pass in the infiltration type instead of setting it here.

Reply:
I agree with these comments. I'll pass in the infil_type into these functions